pom.xml

```diff
@@ -308,6 +313,11 @@
               <artifactId>hadoop-client-runtime</artifactId>
               <version>${hadoop.version}</version>
             </dependency>
+            <dependency>
```

Comment:
@ctubbsii  I am trying to get Fluo to build against Accumulo 2.1.0.  When the ITs are failing because the accumulo servers can't start because they can't find a thrift class.  In the classpath printed out in the accumulo logs I see thrift version 0.12.0.  I tried setting the thrift version here but it still seems to use 0.12.0. Do you have any idea what I am doing wrong?

Reply:
You have to change it at https://github.com/apache/fluo/blob/9e254e2ed69a4991e59c75801b144e1ce964360f/modules/integration-tests/pom.xml#L28-L29

The version set in the main pom is the version that Fluo shades for itself, but the version set in the integration-tests module is expected to be set to the same version as what is being used by the accumulo2-maven-plugin for the version of Accumulo it is supposed to run.

When I set the version there to also be 0.17, the ITs make progress, but I think there might be something wrong with the logging config, because they are very very very noisy on the console. This could have something to do with the switch to log4j2. It could also have something to do with slf4j2 or the fact that ZK uses logback by default for its slf4j binding, or something else along those lines.

Reply:
Thanks @ctubbsii I didnt see that, but I was also on unstable ground when it came to understanding whats going on with Maven here.  Your comment helped me improve my understanding. Are the following stmts correct? Just want to make sure I really understand whats happening.

 1. The accumulo maven plugin needs an accumulo version set (in the plugins dep section) because the plugin will directly call accumulo code to launch miniaccumulo.  When the plugin directly calls miniaccumulo apis it needs to use the 2.1.0 version and not the 2.0.0 version that the plugin was built with.
 2. The plugin launches mini using the classpath of the project (not the plugings classpath).

If the above is correct, then reason num 2 is why setting the thrift version as a plugin dep did not work for me.

Yeah there is definitely something goofy going on with logging.  When I first tried to build the hadoop text class could could not be instantiated because of some logging class not being found.  Bumping the hadoop and zk versions up to something later made things work, not sure why.  I suspect its all revolves around log4j1.

Reply:
I think your statements are essentially correct. I'm a bit fuzzy on the details, having not looked at it in awhile. I think the thrift version property has something to do with the way the minicluster dependency of accumulo2-maven-plugin resolves its own transitive dependencies using Accumulo's pom files, and those pom files use a property to set the version. If Accumulo had hard-coded the thrift version instead of having a property, it probably wouldn't be necessary, but it's been too long for me to remember clearly. I would like to update accumulo2-maven-plugin to work with 2.1 by default, though. Perhaps we can make improvements to make it easier to avoid these kinds of frustrations.

As for logging, I don't know what's going on with Hadoop, but ZK 3.8 uses slf4j 1.7, and defaults to logback. If slf4j 1.7 and logback are excluded as a transitive dependency, using slf4j2, log4j-slf4j2-impl, and log4j-core should work well. That's what Accumulo 2.1 uses, and it should work fine with ZK as well. Fluo currently uses slf4j 1.7 with log4j1, I believe. So, there's a lot that needs to be updated, including Fluo's log4j config files. I could probably help out with some of the logging stuff, once basic functionality is working again with 2.1.

Reply:
in 3121103 I updated to log4j V2 and made changes to get `mvn verify` running end to end.

Reply:
The next step I am going to try is running some test against an accumulo 2.1.0 instance using Fluo

Reply:
>  I would like to update accumulo2-maven-plugin to work with 2.1 by default, though.

It may always be tricky to get that plugin to work w/ a different version of accumulo than it was built with.  Its neat that it worked so well once I got all of the maven config correct.